### PR TITLE
Clean up E0754 explanation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Dylan-DPC 

cc @pickfire

### CODE REVIEW

File: src/librustc_error_codes/error_codes/E0754.md [hunk 1]
```diff
@@ -1,31 +1,27 @@
...
 mod řųśť;
 // ^ error!
-fn main() {}
-```
-
-```compile_fail,E0754
-# #![feature(non_ascii_idents)]
 
 #[no_mangle]
 fn řųśť() {}
```
Comment (pickfire) [inline src/librustc_error_codes/error_codes/E0754.md:12]: Why not make it `error` on the same line like `ok` on the same line as below?

File: src/librustc_error_codes/error_codes/E0754.md [hunk 2]
```diff
@@ -1,31 +1,27 @@
...
 // ^ error!
```
Comment (pickfire) [inline src/librustc_error_codes/error_codes/E0754.md:13]: Why not make it `error` on the same line like `ok` on the same line as below?

```suggestion
fn řųśť() {} // error!
```

File: src/librustc_error_codes/error_codes/E0754.md [hunk 3]
```diff
@@ -1,31 +1,27 @@
...
+
 fn main() {}
 ```
 
-Non-ascii can be used as module names if it is inline
-or a #\[path\] attribute is specified. For example:
+Non-ascii can be used as module names if it is inline or a `#[path]` attribute
```
Comment (pickfire) [inline src/librustc_error_codes/error_codes/E0754.md:18]: `#[path]` attribute? What is that, should we do a link to that page?

Comment (GuillaumeGomez) [reply src/librustc_error_codes/error_codes/E0754.md:18]: Absolutely! As for the `path` attribute, it allows you to link to a file module by its system path and give it a different name.

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/librustc_error_codes/error_codes/E0754.md (hunk 3 above)
Comment (pickfire) [reply src/librustc_error_codes/error_codes/E0754.md]: Yes, but should we link it to the reference to let the users know what is that? I feel like this is a less known item.

Comment (GuillaumeGomez) [reply src/librustc_error_codes/error_codes/E0754.md]: The problem with external links is that they might change (for example in this case, this is: https://doc.rust-lang.org/reference/items/modules.html#the-path-attribute), even if they're part of the rust book.

**Review by pickfire — APPROVED** [review]
Looks good to me.

File: src/librustc_error_codes/error_codes/E0754.md (hunk 3 above)
Comment (pickfire) [reply src/librustc_error_codes/error_codes/E0754.md]: But don't we have link check?

Comment (GuillaumeGomez) [reply src/librustc_error_codes/error_codes/E0754.md]: Only when referring to items inside the docs scope (so relative paths, not absolute ones).

**Pull request merged**